From 0230f41ebd90270be685fc813d23a150afbfc823 Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Fri, 19 Apr 2024 11:01:21 +0800
Subject: [PATCH 012/199] mtk: mt76: mt7996: add beacon_int_min_gcd to support
 different beacon interval

When beacon_int_min_gcd is zero, all beacon intervals for different
interfaces should be same. If beacon_int_min_gcd is larger than zero,
all beacon intervals for different interfaces should be larger or
equal than beacon_int_min_gcd.

Without this patch, set beacon fail when different interfaces use
different beacon interval.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 mt7996/init.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mt7996/init.c b/mt7996/init.c
index 7d8d1e7b..5e969732 100644
--- a/mt7996/init.c
+++ b/mt7996/init.c
@@ -42,6 +42,7 @@ static const struct ieee80211_iface_combination if_comb[] = {
 				       BIT(NL80211_CHAN_WIDTH_40) |
 				       BIT(NL80211_CHAN_WIDTH_80) |
 				       BIT(NL80211_CHAN_WIDTH_160),
+		.beacon_int_min_gcd = 100,
 	}
 };
 
-- 
2.18.0

